File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/startstop/Generator.java
package tasks.startstop;

import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Control;

@JavaTaskInfo(generator=true)
public class Generator extends JavaTask {

	@OutputPortInfo(name = "Out", dataType = Control.class)
	public OutputPort out;
	
	private int i = 4;
	
	@Override public void runnerStart() {
		System.out.println("Generator started");
	}
	
	@Override
	public void generate() {
		if(i-- == 0) super.end();
		out.putData(new Control());
		sleep(500);
	}
	
	@Override public void runnerStop() {
		System.out.println("Generator stopped");
	}
}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/startstop/Generator.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Control import Control

@PythonTaskInfo(generator=True)
class Generator(PythonTask):

    def __init_ports__(self):
        self.out = OutputPort(name="out")

    def runner_start(self):
        print('Generator : start')

    def generate(self):
        self.out.put_data(Control())

    def runner_stop(self):
        print('Generator : stop')
